OpenSceneGraph does not build with GCC 11

wxWidgets packages needed to be rebuilt due to wxWidgets being picky about C++ ABI, but OpenSceneGraph failed to build

```
  D:/a/_temp/msys64/mingw32/i686-w64-mingw32/include/oaidl.h:579:5: error: reference to 'byte' is ambiguous
    579 |     byte *pRecord;
        |     ^~~~
  In file included from D:/a/_temp/msys64/mingw32/include/c++/11.2.0/bits/stl_algobase.h:61,
                   from D:/a/_temp/msys64/mingw32/include/c++/11.2.0/bits/char_traits.h:39,
                   from D:/a/_temp/msys64/mingw32/include/c++/11.2.0/ios:40,
                   from D:/a/_temp/msys64/mingw32/include/c++/11.2.0/ostream:38,
                   from C:/_/mingw-w64-OpenSceneGraph/src/OpenSceneGraph-OpenSceneGraph-3.6.5/include/osg/Notify:20,
                   from C:/_/mingw-w64-OpenSceneGraph/src/OpenSceneGraph-OpenSceneGraph-3.6.5/include/osg/Object:20,
                   from C:/_/mingw-w64-OpenSceneGraph/src/OpenSceneGraph-OpenSceneGraph-3.6.5/include/osg/Matrixd:17,
                   from C:/_/mingw-w64-OpenSceneGraph/src/OpenSceneGraph-OpenSceneGraph-3.6.5/include/osg/DisplaySettings:18,
                   from C:/_/mingw-w64-OpenSceneGraph/src/OpenSceneGraph-OpenSceneGraph-3.6.5/src/osg/DisplaySettings.cpp:13:
  D:/a/_temp/msys64/mingw32/include/c++/11.2.0/bits/cpp_type_traits.h:404:30: note: candidates are: 'enum class std::byte'
    404 |   enum class byte : unsigned char;
        |                              ^~~~
  In file included from D:/a/_temp/msys64/mingw32/i686-w64-mingw32/include/wtypes.h:8,
                   from D:/a/_temp/msys64/mingw32/i686-w64-mingw32/include/winscard.h:10,
                   from D:/a/_temp/msys64/mingw32/i686-w64-mingw32/include/windows.h:97,
                   from C:/_/mingw-w64-OpenSceneGraph/src/OpenSceneGraph-OpenSceneGraph-3.6.5/src/osg/DisplaySettings.cpp:29:
  D:/a/_temp/msys64/mingw32/i686-w64-mingw32/include/rpcndr.h:63:25: note:                 'typedef unsigned char byte'
     63 |   typedef unsigned char byte;
        |                         ^~~~
```

There's probably a `using namespace std;` in there somewhere, otherwise one should be in the `std` namespace and the other in the global namespace

_Originally posted by @jeremyd2019 in https://github.com/msys2/MINGW-packages/issues/9910#issuecomment-954244985_

--- a/mingw-w64-OpenSceneGraph/PKGBUILD
+++ b/mingw-w64-OpenSceneGraph/PKGBUILD
@@ -5,7 +5,7 @@ pkgbase=mingw-w64-${_realname}
 pkgname=("${MINGW_PACKAGE_PREFIX}-${_realname}" "${MINGW_PACKAGE_PREFIX}-${_realname}-debug")
 _pkgver=3.6.5
 pkgver=${_pkgver//-/}
-pkgrel=8
+pkgrel=9
 arch=('any')
 mingw_arch=('mingw32' 'mingw64' 'ucrt64' 'clang64')
 url="http://www.openscenegraph.org/"
@@ -45,13 +45,15 @@ source=(https://github.com/openscenegraph/OpenSceneGraph/archive/${_realname}-${
         find-freetype.patch
         additional-gl-header.patch
         add-gstreamer-cflags.patch
-        las-link-with-boost.patch)
+        las-link-with-boost.patch
+        backport-1055.patch)
 sha256sums=('aea196550f02974d6d09291c5d83b51ca6a03b3767e234a8c0e21322927d1e12'
             '24f64f975e776fc018f9c65a1bcae043dded1d4174342d863f9f205b013acc13'
             '3c7469cf0741905f2c46e4632159220213532d8f194b711b0600b50ad94ab4f8'
             '76eafa53bfc4bb9e0260bf1835622e46210b61cf244a0440beceed9969eeaca7'
             'feb994964fe636e4591394a4c287d27d54cf61c8e5fc5324aebb161cafe2e3dc'
-            '54457dcee1cdd1c486607640e4e0272e619f445291ec4edd04dda99e3c4eb4f7')
+            '54457dcee1cdd1c486607640e4e0272e619f445291ec4edd04dda99e3c4eb4f7'
+            '108bc2ab77959b007dbe5a28be4d06bce364959c89d142a02870f6de466af9bb')
 
 prepare() {
   cd "${srcdir}/${_realname}-${_realname}-${_pkgver}"
@@ -60,6 +62,9 @@ prepare() {
   patch -p1 -i ${srcdir}/additional-gl-header.patch
   patch -p1 -i ${srcdir}/add-gstreamer-cflags.patch
   patch -p1 -i ${srcdir}/las-link-with-boost.patch
+
+  # https://github.com/openscenegraph/OpenSceneGraph/pull/1055
+  patch -p1 -i ${srcdir}/backport-1055.patch
 }
 
 build() {

--- a/mingw-w64-OpenSceneGraph/backport-1055.patch
+++ b/mingw-w64-OpenSceneGraph/backport-1055.patch
@@ -1,0 +1,46 @@
+--- OpenSceneGraph-OpenSceneGraph-3.6.5/src/osg/DisplaySettings.cpp.orig	2020-01-31 12:03:07.000000000 +0100
++++ OpenSceneGraph-OpenSceneGraph-3.6.5/src/osg/DisplaySettings.cpp	2021-10-29 08:18:06.345095100 +0200
+@@ -22,9 +22,6 @@
+ #include <algorithm>
+ #include <string.h>
+ 
+-using namespace osg;
+-using namespace std;
+-
+ #if defined(WIN32) && !defined(__CYGWIN__)
+ #include<windows.h>
+ extern "C" { OSG_EXPORT DWORD NvOptimusEnablement=0x00000001; }
+@@ -32,6 +29,9 @@
+ extern "C" { int NvOptimusEnablement=0x00000001; }
+ #endif
+ 
++using namespace osg;
++using namespace std;
++
+ void DisplaySettings::setNvOptimusEnablement(int value)
+ {
+     NvOptimusEnablement = value;
+diff --git a/src/osgPlugins/cfg/ConfigParser.cpp b/src/osgPlugins/cfg/ConfigParser.cpp
+index d421710043d..8a6807eafce 100644
+--- a/src/osgPlugins/cfg/ConfigParser.cpp
++++ b/src/osgPlugins/cfg/ConfigParser.cpp
+@@ -235,7 +235,6 @@
+ #include "CameraConfig.h"
+ 
+ 
+-using namespace std;
+ using namespace osgProducer;
+ 
+ static void ConfigParser_error( const char * );
+diff --git a/src/osgPlugins/cfg/ConfigParser.y b/src/osgPlugins/cfg/ConfigParser.y
+index cf9adf507ca..f6a28f766ca 100644
+--- a/src/osgPlugins/cfg/ConfigParser.y
++++ b/src/osgPlugins/cfg/ConfigParser.y
+@@ -34,7 +34,6 @@
+ #include <Producer/CameraConfig>
+ 
+ 
+-using namespace std;
+ using namespace Producer;
+ 
+ static void ConfigParser_error( const char * );

